pkg/deploy/controller/lifecycle/manager.go

```diff
@@ -0,0 +1,164 @@
+package lifecycle
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// Status represents the status of a lifecycle action, and should be treated
+// as the source of truth by controllers.
+type Status string
+
+const (
+	// Pending means the action has not yet executed.
+	Pending Status = "Pending"
```

Comment:
@ironcladlou So, this means that the plugins will have to store the logical status somewhere, correct?